packages/auth/src/platform_browser/providers/phone.ts

```diff
@@ -101,11 +101,64 @@ export class PhoneAuthProvider {
    *
    * @returns A Promise for a verification ID that can be passed to
    * {@link PhoneAuthProvider.credential} to identify this flow..
+   *
    */
   verifyPhoneNumber(
     phoneOptions: PhoneInfoOptions | string,
     applicationVerifier: ApplicationVerifier
-  ): Promise<string> {
+  ): Promise<string>;
+
+  /**
+   *
+   * Completes a phone number authentication flow by sending a verification code to the
+   * given phone number, automatically retrieving the verification code from the SMS message,
+   * and signing the user in.
+   *
+   * @example
+   * ```javascript
+   * const provider = new PhoneAuthProvider(auth);
+   * const userCredential = await provider.verifyPhoneNumber(phoneNumber, applicationVerifier, 10);
+   * ```
+   *
+   * @example
+   * An alternative flow is provided using the `signInWithPhoneNumber` method.
+   * ```javascript
+   * const userCredential = signInWithPhoneNumber(auth, phoneNumber, applicationVerifier, 10);
+   * ```
+   *
+   * @param phoneInfoOptions - The user's {@link PhoneInfoOptions}. The phone number should be in
+   * E.164 format (e.g. +16505550101).
+   * @param applicationVerifier - For abuse prevention, this method also requires a
+   * {@link ApplicationVerifier}. This SDK includes a reCAPTCHA-based implementation,
+   * {@link RecaptchaVerifier}.
+   *
+   * @returns A Promise for a UserCredential.
+   */
+  verifyPhoneNumber(
+    phoneOptions: PhoneInfoOptions | string,
+    applicationVerifier: ApplicationVerifier,
+    webOTPTimeout: number
+  ): Promise<UserCredential>;
+
+  verifyPhoneNumber(
+    phoneOptions: PhoneInfoOptions | string,
+    applicationVerifier: ApplicationVerifier,
+    webOTPTimeout?: number
+  ): Promise<unknown> {
```

Comment:
nit: can we explore if "Promise<UserCredential| string>" will work?

Reply:
Yes. This would also work :) I like that it is more specific than using "unknown" - even though this will not be public for developers to see. I have replaced "unknown" in the return types. Please feel free to comment if you believe "unknown" is preferred.